tests/unit/FileHelperTest.php

```diff
@@ -0,0 +1,325 @@
+
+namespace WP2Static;
+
+use Mockery;
+use org\bovigo\vfs\vfsStream;
+use WP_Mock;
+use WP_Mock\Tools\TestCase;
+
+final class FileHelperTest extends TestCase {
+    public function setUp() : void {
+        WP_Mock::setUp();
+    }
+
+    public function tearDown() : void {
+        WP_Mock::tearDown();
+    }
+
+    /**
+     * Test delete_dir_with_files method
+     *
+     * @return void
+     */
+    public function testDeleteDirWithFiles() {
+        // Set up a virual folder structure
+        $structure = [
+            'folder_1' => [
+                'file_1.txt' => 'file_1.txt',
+                'file_2.txt' => 'file_2.txt',
+            ],
+            'folder_2' => [
+                'file_3.txt' => 'file_3.txt',
+                'file_4.txt' => 'file_4.txt',
+                'folder_3' => [
+                    'file_5.txt' => 'file_5.txt',
+                    'file_6.txt' => 'file_6.txt',
+                    
+                ],
+            ],
+        ];
+        $vfs = vfsStream::setup('root');
+        vfsStream::create($structure, $vfs);
+        $filepath = vfsStream::url('root/folder_1');
+
+        // Check vfsStream set up the directories correctly
+        $expected = true;
+        $actual = is_dir( $filepath );
+        $this->assertEquals( $expected, $actual );
+
+        // folder_1 should now be gone
+        FilesHelper::delete_dir_with_files($filepath);
+        $expected = false;
+        $actual = is_dir( $filepath );
+        $this->assertEquals( $expected, $actual );
+
+        // Delete a nested folder
+        $filepath = vfsStream::url('root/folder_2/folder_3');
+        FilesHelper::delete_dir_with_files($filepath);
+        // And confirm the top level one still exists
+        $filepath = vfsStream::url('root/folder_2');
+        $expected = true;
+        $actual = is_dir( $filepath );
+        $this->assertEquals( $expected, $actual );
+    }
+
+    /**
+     * Test getListOfLocalFilesByDir method
+     *
+     * @return void
+     */
+    public function testGetListOfLocalFilesByDir() {
+        // Set up a virual folder structure
+        $structure = [
+            'folder_1' => [
+                'file_1.jpg' => 'file_1.jpg',
+                'file_2.jpg' => 'file_2.jpg',
+            ],
+            'folder_2' => [
+                'file_3.jpg' => 'file_3.jpg',
+                'file_4.jpg' => 'file_4.jpg',
+                'folder_3' => [
+                    'file_5.jpg' => 'file_5.jpg',
+                    'file_6.jpg' => 'file_6.jpg',
+                    
+                ],
+            ],
+        ];
+        $vfs = vfsStream::setup('root');
+        vfsStream::create($structure, $vfs);
+        // Set virtual WP root directory to /root/ for this test
+        $mock = Mockery::mock('overload:\WP2Static\SiteInfo');
+        $mock->shouldreceive('getPath')->andReturn(vfsStream::url('root') . "/");
+
+
+        // Top level folder
+        $filepath = vfsStream::url('root/folder_1');
```

Comment:
How about rather than `folder_1`, `folder_2`, etc - you rename them to `top_level_folder_1`, etc, then can skip the comment lines?

Reply:
Made the structure more descriptive through https://github.com/WP2Static/wp2static/commit/75ebb9a5a4592957bf4354289eb3a8a5adcb563e. If you have any suggestions or require tweaks let me know. Otherwise I think this is good to resolve.